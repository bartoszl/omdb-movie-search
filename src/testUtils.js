import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import theme from './config/theme';

const TestProviders = ({ children }) => (
  <ThemeProvider theme={theme}>
    <MemoryRouter>
      {children}
    </MemoryRouter>
  </ThemeProvider>
);

TestProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) => render(ui, { wrapper: TestProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
