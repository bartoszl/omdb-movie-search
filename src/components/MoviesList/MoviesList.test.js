import React from 'react';
import { render, cleanup } from '../../testUtils';
import MoviesList from './MoviesList';

describe('InfoLabel', () => {
  afterEach(() => {
    cleanup();
  });

  test('matches snapshot', () => {
    const { asFragment } = render((
      <MoviesList />
    ));

    expect(asFragment()).toMatchSnapshot();
  });
});
