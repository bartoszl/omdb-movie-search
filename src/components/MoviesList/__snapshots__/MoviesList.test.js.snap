// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoLabel matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx cfRKE"
  />
</DocumentFragment>
`;
